src/parser/deserializer.ts

```diff
@@ -440,7 +445,9 @@ function deserializeObject(
           _buffer[i] = buffer[index + i];
         }
 
-        if (promoteBuffers && promoteValues) {
+        if (promoteUUIDs && subType === 4) {
+          value = new Binary(_buffer, subType).toUUID();
+        } else if (promoteBuffers && promoteValues) {
           value = _buffer;
```

Comment:
This ordering of logic is important, if promoteBuffers is enabled we will not return a node Buffer anymore if promoteUUIDs is enabled. I think that behavior makes sense and we should note it in the TS-doc for the option.
And then let's also add a test for when both options are passed, assert UUID / Buffer comes out with the combination of true/false for each.

Reply:
just added the tests! with flatMap!